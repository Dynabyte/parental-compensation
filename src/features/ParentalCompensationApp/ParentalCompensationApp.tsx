import React, { Component } from "react";
import TextBoxComponent from "../../utilities/ui/textbox/TextBox";
import { IOptoinProps } from "../../utilities/ui/selectbox/IOptoinProps";
import { SelectBoxComponent } from "../../utilities/ui/selectbox/SelectBox";

import GlobalStyles from "../../utilities/style/GlobalStyle";
import { Container, Row } from "../../utilities/layout";
import Card from "../card/CardComponent";

import { fackData } from "../../fixtures/fackData";

interface IState {
  baseSalary: string;
  data: IOptoinProps[];
}

class ParentalCompensationApp extends Component<any, any> {
  state = {
    baseSalary: "",
    data: [
      { value: "1", text: "one" },
      { value: "2", text: "two" },
      { value: "3", text: "three" }
    ]
  };
  handelChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = event.currentTarget;

    this.setState((prev: IState) => ({ [name]: value }));
  };
  render() {
    return (
      <Container>
        <GlobalStyles />
        <Row Col={2}>
          <TextBoxComponent
            label="GrundlÃ¶n"
            name="baseSalary"
            value={this.state.baseSalary}
            onChange={this.handelChange}
          />

          <SelectBoxComponent
            name="test"
            options={this.state.data}
            label="Test"
          />
        </Row>
        <Row Col={fackData.cardData.length}>
          {fackData.cardData.map((data, index) => (
            <Card key={index} arrData={data} />
          ))}

          <button
            onClick={() => {
              throw new Error("Test .env");
            }}
          >
            Break the world
          </button>
        </Row>
      </Container>
    );
  }
}

export default ParentalCompensationApp;
