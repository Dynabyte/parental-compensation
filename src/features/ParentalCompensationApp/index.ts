export { default as ParentalCompensationApp } from "./ParentalCompensationApp";
