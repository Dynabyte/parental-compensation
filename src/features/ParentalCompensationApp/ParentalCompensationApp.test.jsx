import ParentalCompensationApp from "./ParentalCompensationApp";

describe("Test ParentalCompensationApp Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ParentalCompensationApp />);
  });

  it("should render ParentalCompensationApp correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
