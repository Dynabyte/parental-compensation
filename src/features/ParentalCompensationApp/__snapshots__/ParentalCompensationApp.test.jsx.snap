// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ParentalCompensationApp Component should render ParentalCompensationApp correctly 1`] = `
<styled.div>
  <GlobalStyleComponent />
  <styled.div
    Col={2}
  >
    <TextBoxComponent
      label="GrundlÃ¶n"
      name="baseSalary"
      onChange={[Function]}
      value=""
    />
    <SelectBoxComponent
      label="Test"
      name="test"
      options={
        Array [
          Object {
            "text": "one",
            "value": "1",
          },
          Object {
            "text": "two",
            "value": "2",
          },
          Object {
            "text": "three",
            "value": "3",
          },
        ]
      }
    />
  </styled.div>
  <styled.div
    Col={1}
  >
    <CardComponent
      arrData={
        Object {
          "excessFixedSalary": "53700 kr",
          "max10PBB": "45600 kr",
          "max15PBB90": "697500 kr",
          "parentalSalaryAbove10PBB": "3972,33 kr",
          "parentalSalaryUpto10PBB": "3821,92 kr",
        }
      }
      key="0"
    />
  </styled.div>
</styled.div>
`;
